electron/src/lib/CoreProtocol.ts

```diff
@@ -49,26 +50,30 @@ export class CustomProtocolHandler {
 
     const route = new URL(url);
 
-    switch (route.host) {
-      case ProtocolCommand.START_SSO_FLOW: {
-        if (typeof route.pathname === 'string') {
-          logger.log('Starting SSO flow...');
-          const code = route.pathname.trim().substr(1);
-          await app.whenReady();
-          WindowManager.sendActionAndFocusWindow(EVENT_TYPE.ACCOUNT.SSO_LOGIN, code);
-        }
-        break;
-      }
-      default: {
-        const location = route.href.substr(CORE_PROTOCOL_PREFIX.length);
-        this.hashLocation = `/${location}`;
-        WindowManager.sendActionToPrimaryWindow(EVENT_TYPE.WEBAPP.CHANGE_LOCATION_HASH, this.hashLocation);
-        break;
-      }
+    if (route.host === ProtocolCommand.START_SSO_FLOW) {
+      await this.handleSSOLogin(route);
+    } else {
+      this.forwardHashLocation(route);
+    }
+  }
+
+  private forwardHashLocation(route: URL): void {
+    const location = route.href.substr(CORE_PROTOCOL_PREFIX.length);
+    this.hashLocation = `/${location}`;
+    this.windowManager.sendActionToPrimaryWindow(EVENT_TYPE.WEBAPP.CHANGE_LOCATION_HASH, this.hashLocation);
+  }
+
+  private handleSSOLogin(route: URL): Promise<void> {
+    if (typeof route.pathname === 'string') {
+      logger.log('Starting SSO flow...');
+      const code = route.pathname.trim().substr(1);
+      this.windowManager.sendActionAndFocusWindow(EVENT_TYPE.ACCOUNT.SSO_LOGIN, code).catch(error => {
+        logger.error(`Cannot start SSO flow: ${error.message}`, error);
+      });
     }
   }
 
-  public registerCoreProtocol() {
+  registerCoreProtocol() {
```

Comment:
Everywhere else we are using the `public` keyword. Why remove it now? I think it adds value to the code since it makes it more clear that this function or variable is actually public.

Reply:
I think we can get rid off the `public` keyword because it's the default.

Reply:
Yes, we could, but as I said, I think it adds more value to the code.
Explicit > implicit IMO

Reply:
I agree that the function definitely misses a return type because this one cannot be easily guessed but I would like to avoid defining things as `public` (coding by convention).